### ISSUE TITLE
Q: Any way to provide signerUrl via function or Promise?

### ISSUE THREAD

**Issue by fairbairn** [TTLabs/EvaporateJS#395]
Our signerUrl needs to hit our back-end and is secured by time based and signed requests to our server.

Is there any way to generate the signerUrl via a function instead of a static string?

Since our requests have inherent timeouts, we would need to generate the signerUrl on each request to our back-end to maintain security.  Doing so would also insure our signing logic is secure, otherwise people could hit our back-end URL and sign requests using our secret key.

Changing the signerUrl parameter to accept a function or Promise would provide this capability.

Any ideas on how to implement otherwise?

**Comment by HemalR** [comment]
Yes - except it's not via the signerurl, but you can implement the custom logic you describe via providing a `customAuthMethod` instead.
